File: Module2/coffee.m
% How much caffeine is there in the jar?

% n - number of days
% x - fraction of caffeinated 

nMax = 20; % max number of days to simulate

N = 10; % number of scoops in each jar
x = zeros(1,nMax); % fraction caffeinated
x(1) = 1.0; % initial fraction caffeinated

for n=2:nMax
    
    x(n) = (1-1/N) * x(n-1);
    %x(n) = x(n-1) - 1/N*x(n-1);
    
end % finished loop through days

% THE MODEL ^
% ------------------------------------------
% THE BEHAVIOR / THE OUTPUT ? 

figure(1); 
plot(x,'-ok');
ylabel('fraction caffeinated')
xlabel('Days')

File: Module2/coffee.py
import numpy as np
import matplotlib.pyplot as plt

# How much caffeine is there in the jar?

# n - number of days
# x - fraction of caffeinated 

nMax = 20  # max number of days to simulate

N = 10  # number of scoops in each jar
x = np.zeros(nMax)  # fraction caffeinated
x[0] = 1.0  # initial fraction caffeinated

for n in range(1, nMax):
    x[n] = (1-1/N) * x[n-1]
    #x[n] = x[n-1] - 1/N*x[n-1]
    
# THE MODEL ^
# ------------------------------------------
# THE BEHAVIOR / THE OUTPUT ? 

plt.figure(1)
plt.plot(x, '-ok')
plt.ylabel('fraction caffeinated')
plt.xlabel('Days')
plt.show()
